maintenance/wikia/importMessagingWiki.php

```diff
@@ -0,0 +1,185 @@
+		if ($msg) {
+			# AND page_title LIKE 'Autocreatewiki-welcomebody/%'
+			$sql->AND_('page_title')->LIKE( ucfirst( $msg ) . '/%' );
+		}
+
+		$res = $sql->runLoop( $db, function ( &$res, $row ) use ( &$i, $wgContLang ) {
+			$text = Revision::getRevisionText( $row );
+			$lckey = $wgContLang->lcfirst( $row->page_title );
+			if ( strpos( $lckey, '/' ) ) {
+				$t = explode( '/', $lckey );
+				$key = $t[0];
+				$lang = $t[1];
+			} else {
+				$key = $lckey;
+				$lang = 'en';
+			}
+
+			$res['articlesToDelete'][] = $row->page_title;
+			$res['messages'][$lang] = $res['messages'][$lang] ?? [];
+			$res['messages'][$lang][$key] = $text;
+			$i++;
+		} );
+
+		$this->messages = $res['messages'];
+		$this->articlesToDelete = $res['articlesToDelete'];
+
+		$this->output( "Found $i messages in messaging DB.\n" );
+		if ( $this->isDryRun ) {
+			$this->output( "Running in dry run mode. Existing i18n files will not be changed.\n" );
+		}
+
+		return !empty( $this->messages );
+	}
+
+	/**
+	 * @return array
+	 */
+	private function getMessagesFiles() {
+		global $wgExtensionMessagesFiles;
+
+		return array_unique(
+			array_merge(
+				GlobalMessagesService::getInstance()->getCoreMessageFiles(),
+				GlobalMessagesService::getInstance()->getExtensionMessageFiles(),
+				$wgExtensionMessagesFiles
+			)
+		);
+	}
+
+	/**
+	 * Process all i18n files of enabled extensions, and update them if messaging wiki has a different version
+	 */
+	public function processExtensions() {
+		foreach ( $this->getMessagesFiles() as $extension => $filePath ) {
+			require $filePath;
+			$changed = 0;
+			/** @var array $messages */
+			foreach ( $messages as $lang => $translations ) {
+				if ( !isset( $this->messages[$lang] ) ) {
+					continue;
+				}
+
+				foreach ( $translations as $key => $text ) {
+					if ( isset( $this->messages[$lang][$key] ) &&
+						$this->messages[$lang][$key] !== $messages[$lang][$key]
+					) {
+						$messages[$lang][$key] = $this->messages[$lang][$key];
+						$changed++;
+					}
+				}
+			}
+
+			if ( $changed === 0 ) {
+				# $this->output( "Nothing to update in $extension ($filePath).\n" ); // XXX: DEBUG
+			} else {
+				$this->output( "Found $changed messages to update in $extension ($filePath). Updating...\n" );
+				$this->updateI18nFile( $extension, $filePath, $messages );
+			}
+		}
+	}
+
+	/**
+	 * Regenerate a PHP i18n file
+	 * @param string $extension Extension name
+	 * @param string $filePath i18n file path in app repo
+	 * @param array $messages Messages array in the same format as in i18n files
+	 */
+	protected function updateI18nFile( $extension, $filePath, array $messages ) {
+		$contents = "<?php\n";
+		$contents .= "/** Internationalization file for $extension extension. */\n";
+		$contents .= "\$messages = [];\n\n";
+
+		foreach ( $messages as $lang => $translations ) {
+			$contents .= "\$messages['$lang'] = [\n";
+			foreach ( $translations as $key => $text ) {
+				$contents .= "\t'$key' => '$text',\n";
```

Comment:
We should probably escape apostrophes in message text here to avoid breaking syntax - probably something like `mb_ereg_replace` since we will be dealing with all kinds of characters here.